### DISCUSSION TITLE
regenerating LLVM 10 codegen fixtures

### DISCUSSION THREAD

**Discussion by Birch-san** [bpftrace/bpftrace#1785] [Development]
what's the recommended way to regenerate LLVM 10 codegen fixtures (such as `call_uptr_1_LLVM-10.ll`)? update_codegen_tests.sh focuses on LLVM 8, as far as I can tell?

do I need to install libLLVM 10, wipe my build folder, reconfigure cmake with `-DLLVM_REQUESTED_VERSION=10`, rebuild bpftrace & bpftrace_test, and run it again with `BPFTRACE_UPDATE_TESTS=1`?

is there a better way? my computer's not super fast; rebuilding from scratch take a while.

**Comment by fbs** [comment]
Keep 2 build dirs but yes it is annoying. I only update them when Im 90%+ sure my change is ok.

**Reply by Birch-san** [reply]
okay thanks, yeah; I'll use two build directories then.

**Comment by Birch-san** [comment]
perhaps CI can regenerate codegen fixtures for us?

the [Create Pull Request Action](https://github.com/marketplace/actions/create-pull-request) can be used to [raise a PR which amends your branch](https://github.com/peter-evans/create-pull-request/blob/master/docs/examples.md#use-case-create-a-pull-request-to-modifyfix-pull-requests).

alternatively, a user could request this [by typing `/codegen`](https://github.com/peter-evans/slash-command-dispatch#demos).

the same approach can be used to resolve clang-format, albeit with some tweaking (think you'd not be able to merge the PR into yours, rather you'd hard-reset your branch to match the one CI proposes).

**Reply by fbs** [reply]
Not sure I would like having PRs amend my branch. 

> the same approach can be used to resolve clang-format, albeit with some tweaking (think you'd not be able to merge the PR into yours, rather you'd hard-reset your branch to match the one CI proposes).

For our commit policy you'd have to do that anyway :)

**Reply by Birch-san** [reply]
to clarify: this wouldn't be unsolicited.

you'd raise your PR, the pipeline would fail on codegen.  
a robot would make a separate PR (either automatically, or after you submit a comment `/codegen`). the PR would be one commit ahead of your branch, with regenerated codegen fixtures. you can close if you don't want it, or hit merge to accept the commit into your branch.

ultimately the problem I'm trying to solve is: the CI already knows exactly what codegen change it desires (it shows you a diff!) — how can I get that diff into my branch? ideally the CI would log out a bash one-liner that I can use on my end to apply the changes to my branch.
